explorer/testdata/template/impl_scope.carbon

```diff
@@ -0,0 +1,32 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
```

Comment:
Even with the rest of this PR for context, I have a hard time working out what this test is supposed to validate. I think it's testing support for a template argument depending on a generic binding, but e.g. why attach the template parameter to an `impl forall` rather than a function?

It would probably help to have a more descriptive filename, and maybe some comments and/or more descriptive names in the body of the file.

Reply:
Is this better?